staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -209,30 +209,13 @@ func (c *AvailableConditionController) sync(key string) error {
 		}
 	}
 	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
-	if apiService.Spec.Service != nil && c.serviceResolver != nil {
+	if c.serviceResolver != nil {
 		discoveryURL, err := c.serviceResolver.ResolveEndpoint(apiService.Spec.Service.Namespace, apiService.Spec.Service.Name)
 		if err != nil {
 			return err
 		}
 
-		errCh := make(chan error)
-		go func() {
-			resp, err := c.discoveryClient.Get(discoveryURL.String())
-			if resp != nil {
-				resp.Body.Close()
-			}
-			errCh <- err
-		}()
-
-		select {
-		case err = <-errCh:
-
-		// we had trouble with slow dial and DNS responses causing us to wait too long.
-		// we added this as insurance
-		case <-time.After(6 * time.Second):
```

Comment:
will not hang up, because discoveryClient already set timeout 5s.

Reply:
It didn't for whatever reason. /cc @deads2k

Reply:
> It didn't for whatever reason. /cc @deads2k

Yeah, certain failure modes blackholed in "weird" spots I couldn't find timeouts for.  We chased around a couple and eventually did this to be sure.

Reply:
oh, i thought client.Timeout contains dial and read resonse time. So it must incude dns resolve time. Did i miss something?